components/gitpod-protocol/src/protocol.ts

```diff
@@ -100,6 +100,9 @@ export interface AdditionalUserData {
     ideSettings?: IDESettings;
     // key is the name of the news, string the iso date when it was seen
     whatsNewSeen?: { [key: string]: string }
+    // key is the name of the OAuth client i.e. local app, string the iso date when it was approved
+    // TODO(rl): provide a management UX to allow rescinding of approval
```

Comment:
TIL: _rescinding_ is a synonym for _revoking_ :)

Reply:
Do I understand correctly that we have two levels of approval here: 
1. the OAuth client
2. the tokens issued for a client?

Reply:
Yes.

Given we are using standard OAuth flows, we have:

1. is approval for Gitpod to _issue_ api tokens for a client
2. the actual api tokens issued for a specific client. There may well be more than 1 - different laptops, etc.